models/Stock.go

```diff
@@ -135,40 +136,46 @@ func UpdateStockPrice(stockId uint32, price uint64) error {
 	stock := stockNLock.stock
 	oldStockCopy := *stock
 
-	stock.CurrentPrice = price
-	if price > stock.DayHigh {
-		stock.DayHigh = price
-	} else if price < stock.DayLow {
-		stock.DayLow = price
+	stock.LastTradePrice = price
+
+	stock.UpdatedAt = utils.GetCurrentTimeISO8601()
+
+	// averageStockCount should not be 0, so the math.Max ensures that it is at least 1.
+	// averageStockCount should not be above MAX_AVERAGE_STOCK_COUNT so the math.Min ensures that.
+	// The reason for this upper limit is that if the averageStockCount is too high, the price will never change.
+	averageStockCount := uint64(math.Min(math.Max(float64(STOCK_AVERAGE_PERCENT*(stock.StocksInExchange+stock.StocksInMarket)/100), 1), MAX_AVERAGE_STOCK_COUNT))
+	finalQuantity := uint64(math.Min(float64(quantity), float64(averageStockCount)))
+
+	avgLastPrice.Lock()
+	avgLastPrice.m[stock.Id] += (finalQuantity * (price - avgLastPrice.m[stock.Id])) / averageStockCount
```

Comment:
Like this part seems to be heavy in arithmetic. Could you drop a comment or break this down into functions so that it's a bit more readable?